src/placeos-driver.cr

```diff
@@ -405,9 +425,11 @@ abstract class PlaceOS::Driver
         return metadata if metadata
 
         implements = {{@type.ancestors.map(&.stringify.split("(")[0])}}.reject { |obj| IGNORE_KLASSES.includes?(obj) }
+        iface, funcs = self.functions
```

Comment:
similarly, this method can be this

```crystal
class_getter metadata : String do
  implements = {{@type.ancestors.map(&.stringify.split("(")[0])}}.reject { |obj| IGNORE_KLASSES.includes?(obj) }
  iface, funcs = self.functions

  %({
    "functions": #{self.functions},
    "interface": #{iface},
    "functions": #{funcs},
    "implements": #{implements.to_json},
    "requirements": #{Utilities::Discovery.requirements.to_json},
    "security": #{self.security}
  }).gsub(/\s/, "")
end
```

Reply:
Could `gsub` here accidentally strip spaces in strings?

Reply:
nah as there should not be any spaces in the values (invalid function name with a space etc)